File: Project/features/step_definitions/creditcard_steps.rb

Given(/^I enter a (\d+) number creditcard number$/) do |lengthOfCreditcardNumber|
  @current_page = page(MainPage).await(timeout: 30)

  cardnumber = ""
  while(cardnumber.length < lengthOfCreditcardNumber.to_i)
    cardnumber = "#{cardnumber}9"
  end

  @current_page.enter_creditcard_number(cardnumber)
end

When(/^I press validate$/) do
  @current_page = page(MainPage).await(timeout: 30)
  @current_page.validate
end

Then(/^I should see the succes message$/) do
  @current_page = page(ValidCreditcardPage).await(timeout: 30)
end


Then(/^I should see that the number is to short$/) do
  @current_page = page(MainPage).await(timeout: 30)
end

File: Project/CardValidation.UITest/Steps/CreditCardSteps.cs
using NUnit.Framework;
using System;
using System.Linq;
using System.IO;
using System.Reflection;
using TechTalk.SpecFlow;
using Xamarin.UITest;
using Xamarin.UITest.Queries;
using CreditCardValidation.UITest.Pages;
using System.Globalization;

namespace CreditCardValidation.UITest.Steps
{
    [Binding]
    public class CreditCardSteps
    {
        [Given(@"I enter a (.*) number creditcard number")]
        public void GivenIEnterANumberCreditcardNumber(int lengthOfCreditcardNumber)
        {
            var page = new MainPage().WaitForPage(30) as MainPage;
            page.EnterCreditcardNumer(lengthOfCreditcardNumber);

            Base.App.Screenshot(string.Format(CultureInfo.InvariantCulture, "I enter a {0} number creditcard number", lengthOfCreditcardNumber));
        }

        [When(@"I press validate")]
        public void WhenIPressValidate()
        {
            var page = new MainPage().WaitForPage(30) as MainPage;
            page.PressValidate();

            Base.App.Screenshot(string.Format(CultureInfo.InvariantCulture, "I press validate"));
        }

        [Then(@"I should see the succes message")]
        public void ThenIShouldSeeTheSuccesMessage()
        {
            new ValidCreditcardPage().WaitForPage(30);

            Base.App.Screenshot(string.Format(CultureInfo.InvariantCulture, "I should see the succes message"));
        }
    }
}
